pkgs/servers/sql/postgresql/ext/apache_datasketches.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv, lib, fetchFromGitHub, gcc, postgresql, boost182, nixosTests }:
+let
+  cpp_sources = fetchFromGitHub {
+    owner  = "apache";
+    repo   = "datasketches-cpp";
+    rev    = "refs/tags/4.1.0";
+    sha256 = "sha256-vPoFzRxOXlEAiiHH9M5S6255ahzaKsGNYS0cdHwrRYw=";
+  };
+in
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

Reply:
@drupol Thanks for the hints! Does this look better now?
In particular, what is the right way to resolve the versions for the `cpp_sources` fetch? Does what I made look OK?

Reply:
I added a few more comments about it down here.